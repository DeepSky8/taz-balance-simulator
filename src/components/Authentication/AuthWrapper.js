import React from "react";
import { Link, useNavigate } from "react-router-dom";
import { auth } from "../../firebase/firebase";
import { signOut } from "firebase/auth";



const LoginLink = () => (

    <Link to="/signIn">
        Login
    </Link>

)

const onLogoutClick = (navigate) => {

    signOut(auth)
        .then(navigate("/"))
        .catch((error) => {
            console.log("error thrown on logout:", error)
        })
}

const LogoutLink = () => {
    const navigate = useNavigate()
    return (

        <button onClick={() => {onLogoutClick(navigate) }}> Logout</button >

    )
}

export const AuthWrapper = () => {


    return (
        <div>
            <Link to="/gameSetup/gameInstructions">Game Setup</Link>
            {auth.currentUser.isAnonymous === false
                ?
                <LogoutLink />
                :
                <LoginLink />
            }
        </div>
    )
}

export { AuthWrapper as default }