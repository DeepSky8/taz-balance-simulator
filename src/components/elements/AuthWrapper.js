import React from "react";

export const AuthWrapper = ({children}) => {

    return (
        <div>
            <div>Login</div>
            {children}
        </div>
    )
}

export { AuthWrapper as default }