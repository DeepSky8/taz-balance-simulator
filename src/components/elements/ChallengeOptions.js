import React, { useContext, useEffect, useState } from "react";
import { useReducer } from "react";
// import PickChallengeObject from "./Challenges/PickChallengeObject";
import { villainFlavorArray, villainObjectsArray, villainTitleArray } from "../mission-elements/m-villain";
import { setVillainObject } from "../../actions/challengeSelectActions";
import challengeTransformer from "../../reducers/challengeTransformer";
import PickChallengeObject from './Challenges/PickChallengeObject';

const ChallengeOptions = (
    {
        challengeObjectArray,
        challengeDispatch
    }
) => {

    const onClick = (challengeCode) => {
        // When the button for one of the 
        const challengeObject =
            challengeTransformer(challengeObjectArray, challengeCode)

        challengeDispatch(challengeObject)
    }

    return (
        <div>
            {challengeObjectArray.map((challenge) => {
                return <PickChallengeObject
                    key={challenge.challengeCode}
                    selectChallenge={() => { onClick(challenge.challengeCode) }}
                    {...challenge}
                />
            })}

        </div>
    )

}

export { ChallengeOptions as default }

// <button onClick={selectVillain}>Set villain</button>