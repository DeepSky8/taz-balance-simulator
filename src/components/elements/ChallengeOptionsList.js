import React from "react";
import challengeTransformer from "../../reducers/challengeTransformer";
import PickChallengeObject from './Challenges/PickChallengeObject';

const ChallengeOptionsList = (
    {
        challengeObjectArray,
        challengeDispatch,
        joining
    }
) => (
    <div>
        {challengeObjectArray.map((challenge) => {
            return <PickChallengeObject
                key={challenge.challengeCode}
                selectChallenge={() => { challengeDispatch(challenge) }}
                joining={joining}
                {...challenge}
            />
        })}
    </div>
)

export { ChallengeOptionsList as default }