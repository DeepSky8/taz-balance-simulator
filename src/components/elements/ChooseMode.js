import React from "react";
import { Link } from "react-router-dom";

const ChooseMode = () => {

    return (
        <div>
            <div><Link to='/gameSetup'>Join a game without an account</Link></div>
            <div><Link to='/signIn'>Sign in to host or join a game</Link></div>

        </div>
    )
}

export { ChooseMode as default }