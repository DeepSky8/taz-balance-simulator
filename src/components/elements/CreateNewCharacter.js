import React, { useEffect } from "react";
import { Link, NavLink, Outlet } from "react-router-dom";
import { startSaveNewCharacter } from "../../actions/charActions";

const CreateNewCharacter = ({ newCharState, dispatchNewCharState }) => {

    const saveCharacter = () => {
        startSaveNewCharacter()
    }

    return (
        <div>

            <p>Create New Character page</p>
            <nav>
                <NavLink
                    to='createBard'
                    className={isActive =>
                        (isActive ? "nav-link" : "nav-link-unselected")}
                >Bard</NavLink>
                <NavLink
                    to='createCleric'
                    className={isActive =>
                        (isActive ? "nav-link" : "nav-link-unselected")}
                >Cleric</NavLink>
            </nav>

            <Outlet />



            <p>
                <button onClick={saveCharacter}>Save Character</button>
                <Link to="/gameSetup">Cancel</Link>
            </p>
        </div>
    )
}

export default CreateNewCharacter

// <form id='class-select'>
// <label>Select a class: </label>
// <select
//     name='class'
//     id='class-select'
//     required
// >
//     <option value=''>--Select a class--</option>
//     <option value='bard'>Bard</option>
//     <option value='cleric'>Cleric</option>
//     <option value='rogue'>Rogue</option>
//     <option value='warrior'>Warrior</option>
//     <option value='wizard'>Wizard</option>
// </select>
// </form>


// <div className="dropdown">
// {youArePlayingA}
// <button onClick={toggleClassSelections} className="dropbtn">
//     {newCharState.charClass}
// </button>


// <div id="class-selector" className="dropdown-content">
//     {charClasses.map((charClass) => {
//         return (
//             <div
//                 className="charClass"
//                 key={charClass}
//             >
//                 {charClass}
//             </div>
//         )
//     })}
// </div>
// </div>

// {newCharState.charClass === 'Bard' &&
// <Bard
//     newCharState={newCharState}
//     dispatchNewCharState={dispatchNewCharState}
// />
// }



    // const toggleClassSelections = () => {
    //     document.getElementById('class-selector').classList.toggle('show')
    // }

    // // Close the dropdown if the user clicks outside of it
    // window.onclick = function (event) {
    //     if (!event.target.matches('.dropbtn')) {
    //         var dropdowns = document.getElementsByClassName("dropdown-content");
    //         var i;
    //         for (i = 0; i < dropdowns.length; i++) {
    //             var openDropdown = dropdowns[i];
    //             if (openDropdown.classList.contains('show')) {
    //                 openDropdown.classList.remove('show');
    //             }
    //         }
    //     }
    //     if (event.target.matches('.charClass')) {
    //         console.log('clicked on class: ', event.target.innerText)
    //         const charClass = event.target.innerText
    //         dispatchNewCharState(
    //             setCharClass(
    //                 charClass
    //             )
    //         )
    //     }
    // }


    // useEffect(() => {
    //     const charSelector = document.getElementById('class-select')

    //     charSelector.addEventListener('change', (e) => {
    //         const charClass = e.target.value

    //         dispatchNewCharState(
    //             setCharClass(
    //                 charClass
    //             )
    //         )
    //     })
    // }, [])

