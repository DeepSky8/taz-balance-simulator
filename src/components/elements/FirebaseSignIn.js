import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { auth, ui } from '../../firebase/firebase';
import { uiConfig } from '../../firebase/uiConfig';
import FirebaseAuth from 'react-firebaseui/FirebaseAuth';


const FirebaseSignIn = () => {

    // useEffect(() => {
    //     if (ui.isPendingRedirect()) {
    //         ui.start('#firebaseui-auth-container', uiConfig);
    //     } else {
    //         ui.start('#firebaseui-auth-container', uiConfig);
    //     }
    // }, [])

    return (
        <div>
            Firebase UI Sign In Page
            <Link to="/gameSetup">Back</Link>

            <FirebaseAuth uiConfig={uiConfig} firebaseAuth={auth} />
        </div>

    )
}
export { FirebaseSignIn as default }

// <div id="firebaseui-auth-container">

// </div>