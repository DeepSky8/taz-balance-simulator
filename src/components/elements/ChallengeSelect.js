import React, { useEffect, useReducer, useState } from "react";
import {
    setVillainObject,
    setRelicObject,
    setLocationObject,
    toggleLocation,
    toggleRelic,
    toggleVillain,
    startSetChallenges
} from "../../actions/challengeSelectActions";
import { challengeSelectReducer, defaultChallengeState } from "../../reducers/challengeSelectReducer";
import ChallengeOptionsList from "./ChallengeOptionsList";
import { villainObjectsArray } from "./Challenges/mission-elements/m-villain";
import { relicObjectsArray } from "./Challenges/mission-elements/m-relic";
import { locationObjectsArray } from "./Challenges/mission-elements/m-location";

const ChallengeSelect = ({ setupState, dispatchSetupState }) => {
    const [challengeState, dispatchChallengeState] = useReducer(challengeSelectReducer, defaultChallengeState)
    const [challengesUpdated, updateChallenge] = useState(0)
    const dispatchedChallengeCode = () => {
        updateChallenge(challengesUpdated + 1)
    }
    const uid = setupState.uid
    const gameID = setupState.gameID

    const toggleVillainDisplay = () => {
        dispatchChallengeState(toggleVillain())
    }
    const toggleRelicDisplay = () => {
        dispatchChallengeState(toggleRelic())
    }
    const toggleLocationDisplay = () => {
        dispatchChallengeState(toggleLocation())
    }

    const villainDispatch = (challengeObject) => {
        if (!setupState.joiningGame) {

            // const challengeObject =
            //     challengeTransformer(villainObjectsArray, challengeCode)
            dispatchChallengeState(setVillainObject(challengeObject))
            dispatchedChallengeCode()
        }
    }

    const relicDispatch = (challengeObject) => {
        if (!setupState.joiningGame) {

            // const challengeObject =
            //     challengeTransformer(relicObjectsArray, challengeCode)
            dispatchChallengeState(setRelicObject(challengeObject))
            dispatchedChallengeCode()
        }
    }

    const locationDispatch = (challengeObject) => {
        if (!setupState.joiningGame) {

            // const challengeObject =
            //     challengeTransformer(locationObjectsArray, challengeCode)
            dispatchChallengeState(setLocationObject(challengeObject))
            dispatchedChallengeCode()
        }
    }

    useEffect(() => {
        startSetChallenges(
            {
                villainCode: challengeState.selectedVillainObject.challengeCode,
                relicCode: challengeState.selectedRelicObject.challengeCode,
                locationCode: challengeState.selectedLocationObject.challengeCode
            },
            uid
        )
    }, [challengesUpdated])

    return (
        <div>
            {!setupState.joiningGame &&
                <div>
                    Battle
                    <button onClick={toggleVillainDisplay}>
                        {challengeState.selectedVillainObject.challengeName}
                    </button>
                    to reclaim
                    <button onClick={toggleRelicDisplay}>
                        {challengeState.selectedRelicObject.challengeName}
                    </button>

                    {challengeState.selectedLocationObject.challengeCode === 'l3'
                        ?
                        'on'
                        :
                        'in'
                    }

                    <button onClick={toggleLocationDisplay}>
                        {challengeState.selectedLocationObject.challengeName}
                    </button>
                </div>
            }

            {setupState.joiningGame &&
                <div>
                    Your party will battle
                    <button onClick={toggleVillainDisplay}>
                        {challengeState.selectedVillainObject.challengeName}
                    </button>
                    to reclaim
                    <button onClick={toggleRelicDisplay}>
                        {challengeState.selectedRelicObject.challengeName}
                    </button>

                    {challengeState.selectedLocationObject.challengeCode === 'l3' ?
                        'on'
                        :
                        'in'
                    }

                    <button onClick={toggleLocationDisplay}>
                        {challengeState.selectedLocationObject.challengeName}
                    </button>
                </div>
            }


            <div>
                {challengeState.villainDisplayed &&
                    <ChallengeOptionsList
                        challengeObjectArray={villainObjectsArray}
                        challengeDispatch={villainDispatch}
                        joining={setupState.joiningGame}
                    />}

                {challengeState.relicDisplayed &&
                    <ChallengeOptionsList
                        challengeObjectArray={relicObjectsArray}
                        challengeDispatch={relicDispatch}
                        joining={setupState.joiningGame}
                    />}

                {challengeState.locationDisplayed &&
                    <ChallengeOptionsList
                        challengeObjectArray={locationObjectsArray}
                        challengeDispatch={locationDispatch}
                        joining={setupState.joiningGame}
                    />}
            </div>
        </div>
    )
}

export { ChallengeSelect as default }

// startSetChallenges(challengeCode, auth.currentUser.uid, )

