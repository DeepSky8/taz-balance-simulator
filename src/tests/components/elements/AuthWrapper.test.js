import React from "react";
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { AuthWrapper } from '../../../components/elements/AuthWrapper';

test('should render the Authentication Wrapper', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AuthWrapper />)
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})