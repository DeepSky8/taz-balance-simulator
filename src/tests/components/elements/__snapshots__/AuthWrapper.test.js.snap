// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Authentication Wrapper 1`] = `
<div>
  <div>
    Login
  </div>
</div>
`;
